bonsai-core/src/main/scala/com/stripe/bonsai/layout/Product3Layout.scala

```diff
@@ -0,0 +1,95 @@
+package com.stripe.bonsai
+package layout
+
+import java.io.{ DataOutput, DataInput }
+
+class Product3Layout[A, B, C, D](
```

Comment:
Yeah, probably. Though it may make more sense to actually create a macro for these instead that'll work with case classes.